### PR TITLE
Add vagrant as a deployment 'target' for romana-setup tool

### PR DESCRIPTION
**Pull request by cgilmour**
Quite a few changes made, reorganizing things and so on, so that the tool works "nicely" with both AWS and Vagrant as targets.

It has been extensively tested, but apart from the README, not much in terms of review.
I'd appreciate a quick review, but we do want this to hit master fairly promptly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,36 +71,90 @@ Run the installer. This will create the Devstack cluster, install and activate R
...
 The EC2 installation takes 20-25 mins to complete (on t2.large instances) and creates an OpenStack DevStack cluster (Liberty release) with a single Controller Node and up to 4 Compute Nodes. Each OpenStack Node runs on a dedicated EC2 instance. Romana installs its OpenStack ML2 and IPAM drivers and creates a Romana router gateway interface on each Compute Node.
 
-Once installed, you can perform a variety of checks and experiments on your own. See the [Wiki](https://github.com/romana/romana/wiki) for details.
+
+### Teardown
 
 To uninstall, use the 'uninstall' subcommand.
 ```bash
 ./romana-setup uninstall
 ```
```
Comment (jbrendel) [inline README.md]: Add a note or link here that the reader should jump to the "Using" section?

File: README.md [hunk 2]
```diff
@@ -53,7 +54,7 @@ Ensure the IAM user or role for these credentials has permission to create these
 This will be addressed in the future.
 
 
-### Install
+### Installation
 
 Check out Romana repository.
 ```bash
```
Comment (jbrendel) [inline README.md]: Above you used "sh-session" as formatting aide, but here it's "bash"?

Comment (cgilmour) [inline README.md]: 'sh-session' for copy-paste from a terminal session (including prompt and command output)
'bash' for plain commands that a user should execute.

File: README.md (hunk 1 above)
Comment (cgilmour) [inline README.md]: I'd put it just after the ./romana-setup bit.

File: README.md [hunk 3]
```diff
@@ -70,36 +71,90 @@ Run the installer. This will create the Devstack cluster, install and activate R
...
+And [contact us](#getting-help) for more information or assistance.
+
+*Note:* By default, the stack name will be your username.
+If you prefer to use a different name for the stack, you can override this by providing an extra option when launching:
+```bash
+./romana-setup install -e stack_name=xyz
+```
+The name should be a single word, and only contain letters and numbers. (No hyphens, underscores, etc.)
+
 The EC2 installation takes 20-25 mins to complete (on t2.large instances) and creates an OpenStack DevStack cluster (Liberty release) with a single Controller Node and up to 4 Compute Nodes. Each OpenStack Node runs on a dedicated EC2 instance. Romana installs its OpenStack ML2 and IPAM drivers and creates a Romana router gateway interface on each Compute Node.
```
Comment (jbrendel) [inline README.md]: Add a note on how to get more or less compute nodes and how many there are by default? Two compute nodes, on which also acts as controller.

File: README.md [hunk 4]
```diff
@@ -70,36 +71,90 @@ Run the installer. This will create the Devstack cluster, install and activate R
...
 
-Once installed, you can perform a variety of checks and experiments on your own. See the [Wiki](https://github.com/romana/romana/wiki) for details.
+
+### Teardown
 
 To uninstall, use the 'uninstall' subcommand.
 ```bash
 ./romana-setup uninstall
```
Comment (jbrendel) [inline README.md]: Does this actually destroy the stack? Or the Vagrant VMs if this is for Vagrant?

File: README.md [hunk 5]
```diff
@@ -70,36 +71,90 @@ Run the installer. This will create the Devstack cluster, install and activate R
...
+```sh-session
+$ type -P vagrant
+/usr/local/bin/vagrant
+```
 
-### Use
+### Installation
 
-- connect to your host: `ssh -i ~/.ssh/ec2_id_rsa ubuntu@controller-ip`
+Check out Romana repository.
```
Comment (jbrendel) [inline README.md]: "Check out THE Romana repository." ?

File: README.md (hunk 4 above)
Comment (cgilmour) [inline README.md]: It does.

File: romana-install/roles/romana-common/tasks/network.yml [hunk 6]
```diff
@@ -1,4 +1,12 @@
 ---
+# Remove the virbr0 interface
```
Comment (jbrendel) [inline romana-install/roles/romana-common/tasks/network.yml]: Let's add a comment here about why we remove it: It's an artefact of the Devstack install and we don't need bridges or anything like it.

File: romana-install/vagrant.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
```
Comment (jbrendel) [inline romana-install/vagrant.py]: Since we explicitly create VMs with specific IP addresses, why do we need a Vagrant inventory script?

**Comment by jbrendel** [comment]
Looks good. Let's merge it.

**Pull request merged**